src/Execution/MutationExecutor.php

```diff
@@ -304,6 +436,16 @@ public static function executeUpdate(Model $model, Collection $args, ?Relation $
                 });
             }
 
+            if (isset($nestedOperations['upsert'])) {
+                (new Collection($nestedOperations['upsert']))->each(function ($singleValues) use ($relation): void {
```

Comment:
That iterations has been duplicated a few times, it could be abstracted into its own function, like `handleMultiRelationUpsert`

Reply:
I used the same approach than update. Should I abstract the update too?